🧑‍🌾  Element.cc warning in gz-sim10 and gz-sim-main and gz-launch windows

## Environment
* OS Version: cnlwin
* Source sdformat build

## Description
* Expected behavior: Build should pass
* Actual behavior: Build is unstable with warnings

## Steps to reproduce
1. Run a build in [gz-sim-cnlwin](https://build.osrfoundation.org/job/gz_sim-10-cnlwin/)
2. See warning appear

## Output
Reference build:
- https://build.osrfoundation.org/job/gz_sim-10-cnlwin/23/ (the build actually passes but you can see the warning)
- https://build.osrfoundation.org/job/gz_launch-main-cnlwin/58/(same beahvior, the build passes but you can see the warning)


Warning:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include\optional(69,17): warning C4244: 'initializing': conversion from '_Ty' to 'unsigned int', possible loss of data [C:\J\workspace\gz_sim-10-cnlwin\ws\build\sdformat\src\sdformat.vcxproj]

          with

          [

              _Ty=__int64

          ] (compiling source file C:\J\workspace\gz_sim-10-cnlwin\ws\src\sdformat\src\Element.cc)
```

happening for multiple files in sdformat

--- a/src/Element.cc
+++ b/src/Element.cc
@@ -699,7 +699,7 @@ std::optional<unsigned int> ElementPrivate::ElementDescriptionIndex(
   {
     if ((*iter)->GetName() == _key)
     {
-      return std::static_cast<unsigned int>(
+      return static_cast<unsigned int>(
         std::distance(this->elementDescriptions.begin(), iter));
     }
   }

--- a/src/Element.cc
+++ b/src/Element.cc
@@ -699,7 +699,8 @@ std::optional<unsigned int> ElementPrivate::ElementDescriptionIndex(
   {
     if ((*iter)->GetName() == _key)
     {
-      return std::distance(this->elementDescriptions.begin(), iter);
+      return std::static_cast<unsigned int>(
+        std::distance(this->elementDescriptions.begin(), iter));
     }
   }
   return std::nullopt;

